tests/components/frontier_silicon/test_config_flow.py

```diff
@@ -0,0 +1,266 @@
+        CONF_PIN: "1234",
+    }
+    mock_setup_entry.assert_called_once()
+
+
+async def test_form_nondefault_pin(
+    hass: HomeAssistant, mock_setup_entry: AsyncMock
+) -> None:
+    """Test we get the form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch(
+        "afsapi.AFSAPI.get_friendly_name",
+        side_effect=InvalidPinException,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "1.1.1.1", CONF_PORT: 80},
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["step_id"] == "device_config"
+    assert result2["errors"] is None
+
+    result3 = await hass.config_entries.flow.async_configure(
+        result2["flow_id"],
+        {CONF_PIN: "4321"},
+    )
+    await hass.async_block_till_done()
+
+    assert result3["type"] == FlowResultType.CREATE_ENTRY
+    assert result3["title"] == "Name of the device"
+    assert result3["data"] == {
+        "webfsapi_url": "http://1.1.1.1:80/webfsapi",
+        "pin": "4321",
+    }
+    mock_setup_entry.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    ("friendly_name_error", "result_error"),
+    [
+        (ConnectionError, "cannot_connect"),
+        (InvalidPinException, "invalid_auth"),
+        (ValueError, "unknown"),
+    ],
+)
+async def test_form_nondefault_pin_invalid(
+    hass: HomeAssistant, friendly_name_error: Exception, result_error: str
+) -> None:
+    """Test we get the proper errors when trying to validate an user-provided PIN."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {}
+
+    with patch(
+        "afsapi.AFSAPI.get_friendly_name",
+        side_effect=InvalidPinException,
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {CONF_HOST: "1.1.1.1", CONF_PORT: 80},
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["step_id"] == "device_config"
+    assert result2["errors"] is None
+
+    with patch(
+        "afsapi.AFSAPI.get_friendly_name",
+        side_effect=friendly_name_error,
+    ):
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {CONF_PIN: "4321"},
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == FlowResultType.FORM
+    assert result2["step_id"] == "device_config"
+    assert result3["errors"] == {"base": result_error}
```

Comment:
This test is incomplete. Please finish the test with the user doing an second attempt and finishes the config flow until the create entry.

This ensures there are no side-effects from the error.